sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java

```diff
@@ -415,6 +471,140 @@ public BoundedWindow window() {
   }
 
   /**
+   * Annotation for the method that maps an element to an initial restriction for a <a
+   * href="https://s.apache.org/splittable-do-fn">splittable</a> {@link DoFn}.
+   *
+   * <p>Signature: {@code RestrictionT getInitialRestriction(InputT element);}
```

Comment:
I'd prefer not to clutter this PR with more convenience code such as optional parameters - I'd rather leave that for later when the rest of it works.